src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityMultiFurnace.java

```diff
@@ -198,9 +205,12 @@ protected Recipe findRecipe(long maxVoltage,
             }
 
             // If there were no accepted ingredients, then there is no recipe to process.
-            if(recipeInputs.isEmpty()) {
+            // the output may be filled up
+            if (recipeInputs.isEmpty() && !invalidInputsForRecipes) {
```

Comment:
The only case where `!invalidInputsForRecipes` would be true is when a matching recipe exists, and so the recipe inputs would not be empty, so this logic doesn't quite make sense.

Reply:
if the output is full ( so canFit is false ) for that recipe, we exit the loop before adding to recipeInputs.
in that case, an empty recipeInputs does not mean that there are no valid input, it means we could not add anything because the output is full.

Reply:
However, we could have iterated through the for loop once first, which would set the state of `invalidInputsForRecipes`. It might be a good idea to reset the value of `invalidInputsForRecipes` if the check for `canFitOutputs` passes, but before the break out of the loop.